src/tyr/route_serializer_osrm.cc

```diff
@@ -325,6 +334,37 @@ struct IntersectionEdges {
   }
 };
 
+// Forward declaration
+valhalla::baldr::json::MapPtr serialize_incident(const TripLeg_Node_Incident& incident);
+
+// Serializes incidents and adds to json-document
+void addsIncidents(
+    const google::protobuf::RepeatedPtrField<valhalla::TripLeg_Node_Incident>& incidents,
+    json::Jmap& doc) {
+  if (incidents.size() == 0) {
+    // No incidents, nothing to do
+    return;
+  }
+  json::ArrayPtr serialized_incidents = std::shared_ptr<json::Jarray>(new json::Jarray());
+  for (const auto& incident : incidents) {
+    {
```

Comment:
Whats the purpose of this additional scope?

Reply:
Keeping the local context small on the temporary variables used for bringing out any existing array in the json doc.